packages/action/src/models/action-event-utils.ts

```diff
@@ -0,0 +1,324 @@
+  /**
+   * Name assigned to the Action
+   */
+  actionName: string;
+  /**
+   * Id of the the current Action run
+   */
+  actionRunId: string;
+  /**
+   * Previous Action run information
+   */
+  previousRun?: PreviousActionRunInfo;
+}
+
+export type MonitorConfirmation = number | 'safe' | 'finalized';
+
+export interface PreviousActionRunInfo {
+  trigger: 'schedule' | 'webhook' | 'monitor' | 'monitor-filter' | 'scenario' | 'manual' | 'manual-api';
+  status: 'pending' | 'throttled' | 'error' | 'success';
+  createdAt: string;
+  actionId: string;
+  message?: string;
+  actionRunId: string;
+}
+
+/**
+ * Key-value secrets defined in the Action secrets vault
+ */
+export interface ActionSecretsMap {
+  [k: string]: string;
+}
+
+/**
+ * Action request data injected by Defender based on the type of trigger
+ */
+export interface ActionRequestData {
+  /**
+   * Main payload of the request: can be a generic object for a webhook request, or a typed request from Monitors
+   */
+  body?: MonitorConditionRequest | MonitorTriggerEvent | Record<string, unknown>;
+
+  /**
+   * Query parameters of the webhook request
+   */
+  queryParameters?: { [name: string]: string };
+
+  /**
+   * HTTP headers parameters of the webhook request (only headers starting with 'X-' are received)
+   */
+  headers?: { [name: string]: string };
+}
+
+/**
+ * Payload injected by a Monitor when using an Action as a condition.
+ *
+ * Note that the Action should return a MonitorConditionResponse corresponding to the matched txs.
+ */
+export interface MonitorConditionRequest {
+  /**
+   * All potential matches to be evaluated by the action
+   */
+  events: MonitorTriggerEvent[];
+}
+
+/**
+ * To be returned by an Action when invoked as a Monitor condition via a MonitorConditionRequest to refine a match
+ */
+export interface MonitorConditionResponse {
+  /**
+   * List of matches to be triggered by the Monitor, must be included in the MonitorConditionRequest
+   */
+  matches: MonitorConditionMatch[];
+}
+
+/**
+ * Match to be triggered by the Monitor
+ */
+export interface MonitorConditionMatch {
+  /**
+   * Hash of the transaction to match
+   */
+  hash: string;
+
+  /**
+   * Optional user-defined metadata to include with the Monitor notification
+   */
+  metadata?: { [k: string]: unknown };
+}
+
+/**
+ * Represents an object matched by a monitor
+ */
+export type MonitorTriggerEvent = BlockTriggerEvent | FortaTriggerEvent;
+
+export interface BlockTriggerEvent {
+  type: 'BLOCK';
+  hash: string;
+  timestamp: number;
+  blockNumber: string;
+  blockHash: string;
+  transaction: EthReceipt;
+  matchReasons: MonitorConditionSummary[];
+  matchedAddresses: string[];
+  matchedChecksumAddresses: string[];
+  sentinel: BlockMonitorSummary;
+  metadata?: { [k: string]: unknown };
+}
+
+export interface FortaTriggerEvent {
+  type: 'FORTA';
+  hash: string;
+  alert: FortaAlert;
+  matchReasons: FortaConditionSummary[];
+  sentinel: FortaMonitorSummary;
```

Comment:
I didn't change this by `monitor` because it would require a big backend refactor to keep it backwards compatible